packages/components/.storybook/main.js

```diff
@@ -9,25 +9,65 @@ function getAbsolutePath(value) {
 }
 
 module.exports = {
-    stories: ['../src/**/*.stories.*'],
+    stories: ['../src/**/*.@(mdx|stories.*)'],
     logLevel: 'debug',
+
     addons: [
         getAbsolutePath('@storybook/addon-links'),
         getAbsolutePath('@storybook/addon-controls'),
         getAbsolutePath('@storybook/addon-viewport'),
         getAbsolutePath('@storybook/addon-actions'),
     ],
+
     staticDirs: ['../public', { from: '../../suite-data/files', to: '/static' }],
+
     framework: {
-        name: getAbsolutePath('@storybook/react-webpack5'),
+        name: getAbsolutePath('@storybook/nextjs'),
         options: {},
     },
-    babel: options => {
-        options.presets.push('@babel/preset-typescript');
 
-        return options;
+    typescript: {
+        check: false, // Disable type checking
+        reactDocgen: 'react-docgen-typescript',
+        reactDocgenTypescriptOptions: {
+            shouldExtractLiteralValuesFromEnum: true,
+            propFilter: prop => (prop.parent ? !/node_modules/.test(prop.parent.fileName) : true),
+            skipChildrenPropWithoutDoc: false,
+        },
     },
-    features: {
-        storyStoreV7: false, // Remove this line when storiesOf is not used anymore
+
+    webpackFinal: config => {
+        // Add TypeScript support
+        config.module.rules.push({
+            test: /\.tsx?$/,
+            use: [
+                {
+                    loader: require.resolve('babel-loader'),
+                    options: {
+                        presets: [
+                            '@babel/preset-env',
+                            ['@babel/preset-react', { runtime: 'automatic' }],
+                            '@babel/preset-typescript',
+                        ],
+                    },
+                },
+            ],
+        });
+
+        // NOTE: remove the previous loaders from handling the svgs
+        const imageRule = config.module.rules.find(rule => rule?.['test']?.test('.svg'));
+        if (imageRule) {
+            imageRule['exclude'] = /\.svg$/;
+        }
+
+        // Configure SVG files to match the main project's webpack config
+        config.module.rules.push({
+            test: /\.(gif|jpe?g|png|svg)$/,
```

Comment:
this is the same config we have in the proejct and with it, it laods the coin icons properly